javascript/javascript1/week4/voice_assistant.js

```diff
@@ -1,68 +1,81 @@
-// *Hello my name is Benjamin - Should save the name benjamin. and respond with "nice to meet you Benjamin". What if someone writes this twice?
-// *What is my name - should respond with the name of the person. What if the name has not yet been mentioned?
-// *Add fishing to my todo - Should respond with "fishing added to your todo". Should add fishing to a list of todos
-// *Add singing in the shower to my todo - Should add singing in the shower to a list of todos
-// *Remove fishing from my todo - Should respond with "Removed fishing from your todo"
-// *What is on my todo? - should respond with the todos. Fx you have 2 todos - fishing and singing in the shower
-// *What day is it today? - Should respond with the date in a human readable format. E.g. if today is 30/8/2019 then it should respond with 30. of August 2019
-// Should be able to do simple math. fx what is 3 + 3 should respond with 6. Or what is 4 * 12 should respond with 48
-// Set a timer for 4 minutes - Should respond with "Timer set for 4 minutes". When 4 minutes is up: "Timer done". How do we set a timer in js? Google is your friend here!
-// Add one or more command to your voice assistant
+const state = {
+  // initial values
+  userName: null,
+  toDoList: [],
+};
+const patterns = {
+  namePattern: /Hello, my name is (.+)/i,
+  addPattern: /Add (.+) to my to do list\./i,
+  removePattern: /Remove (.+) from my to do list\./i,
+  showToDoPattern: /What is my to do\?/,
+  mathPattern: /what is (\d+)\s*([+\-*\/])\s*(\d+)\?/i,
+  tempPattern: /It is (\d+) degrees. What do I wear\?/,
+  getNamePattern: /What is my name\?/i,
+  datePattern: /What day is it today\?/i,
+  timerPattern: /Set a timer for (\d+) min/i,
+};
+const commands = {
+  [patterns.namePattern]: {
+    reply: (input) => {
+      const userName = getUserName(input);
+      if (userName) {
+        state.userName = userName;
+        return `Nice to meet you, ${state.userName}`;
+      } else {
+        return "Sorry, I could not determine your name.";
+      }
+    },
+  },
+  [patterns.addPattern]: {
+    reply: (input) => {
+      return addToDo(input);
+    },
+  },
+  [patterns.removePattern]: {
+    reply: (input) => {
+      return removeToDo(input);
+    },
+  },
```

Comment:
Very minor syntactic adjustment here but be as consistent. Below you write the functions without the curly brackets and return  { return } and here with. There's arguments to have have either but consistency is key.